code/game/gamemodes/cult/talisman.dm

```diff
@@ -4,113 +4,113 @@
 	var/uses = 0
 	info = "<center><img src='talisman.png'></center><br/><br/>"
 
-	attack_self(mob/living/user as mob)
-		if(iscultist(user))
-			var/delete = 1
-			// who the hell thought this was a good idea :(
-			switch(imbue)
-				if("newtome")
-					call(/obj/effect/rune/proc/tomesummon)()
-				if("armor")
-					call(/obj/effect/rune/proc/armor)()
-				if("emp")
-					call(/obj/effect/rune/proc/emp)(usr.loc,3)
-				if("conceal")
-					call(/obj/effect/rune/proc/obscure)(2)
-				if("revealrunes")
-					call(/obj/effect/rune/proc/revealrunes)(src)
-				if("ire", "ego", "nahlizet", "certum", "veri", "jatkaa", "balaq", "mgar", "karazet", "geeri")
-					call(/obj/effect/rune/proc/teleport)(imbue)
-				if("communicate")
-					//If the user cancels the talisman this var will be set to 0
-					delete = call(/obj/effect/rune/proc/communicate)()
-				if("deafen")
-					call(/obj/effect/rune/proc/deafen)()
-				if("blind")
-					call(/obj/effect/rune/proc/blind)()
-				if("runestun")
-					user << "<span class='warning'>To use this talisman, attack your target directly.</span>"
-					return
-				if("supply")
-					supply()
-			user.take_organ_damage(5, 0)
-			if(src && src.imbue!="supply" && src.imbue!="runestun")
-				if(delete)
-					qdel(src)
-			return
-		else
-			user << "You see strange symbols on the paper. Are they supposed to mean something?"
-			return
-
-
-	attack(mob/living/carbon/T as mob, mob/living/user as mob)
-		if(iscultist(user))
-			if(imbue == "runestun")
-				user.take_organ_damage(5, 0)
-				call(/obj/effect/rune/proc/runestun)(T)
+/obj/item/weapon/paper/talisman/attack_self(mob/living/user as mob)
+	if(iscultist(user))
+		var/delete = 1
+		// who the hell thought this was a good idea :(
+		switch(imbue)
+			if("newtome")
+				call(/obj/effect/rune/proc/tomesummon)(user)

```

Comment:
This _really_ should not be using call like this. Perhaps refactor these procs into a cult datum that both runes and talismans have as a var.